sklearn/metrics/_pairwise_distances_reduction/_base.pyx.tp

```diff
@@ -358,7 +354,18 @@ cdef class BaseDistanceReducer{{name_suffix}}:
         ITYPE_t X_start,
         ITYPE_t X_end,
     ) nogil:
-        """Initialize datastructures used in a thread given its number."""
+        """Initialize datastructures used in a thread given its number.
+
+        EuclideanDistance specialisations of subclass of BaseDistancesReduction _must_
+        in this method call:
```

Comment:
Nit: For clarification:

```suggestion
        In this method, EuclideanDistance specialisations of subclass of 
        BaseDistancesReduction _must_ call:
```

This comments applies to the other docstrings as well.